modules/noise/noise.h

```diff
@@ -275,20 +276,24 @@ class Noise : public Resource {
 		return out.l;
 	}
 
+	GDVIRTUAL1RC(real_t, _get_noise_1d, real_t)
+	GDVIRTUAL1RC(real_t, _get_noise_2d, Vector2)
+	GDVIRTUAL1RC(real_t, _get_noise_3d, Vector3)
```

Comment:
Should these be required? Or, is there a case where you'd make a sub-class but not implement them?

Making them required would be:

```suggestion
	GDVIRTUAL1RC_REQUIRED(real_t, _get_noise_1d, real_t)
	GDVIRTUAL1RC_REQUIRED(real_t, _get_noise_2d, Vector2)
	GDVIRTUAL1RC_REQUIRED(real_t, _get_noise_3d, Vector3)
```

Reply:
Hmm I don't know. Personally I can implement them all in my project, but I don't know if it would be annoying in other situations.

Would this macro end up applying to FastNoiseLite, which is a derived core class from which the methods are already implemented but in C++ and therefore not by gdvirtual?